### DISCUSSION TITLE
Classic queue alternative in RabbitMQ 4

### DISCUSSION THREAD

**Discussion by ioutkine** [rabbitmq/rabbitmq-server#11286] [Questions] [answered]
There are indeed valid scenarios where temporary classic queues are the optimal choice. These include, but are not limited to, RPC  request-response patterns and applications that dynamically create queues to monitor events during runtime. Given the impending deprecation of classic queues in future RabbitMQ versions, it's natural to question whether it's time to explore other messaging solutions or there is hope that RabbitMQ will introduce alternatives to accommodate these use cases?

**Comment by mkuratczyk** [comment, marked as answer]
Classic queues are not deprecated. In fact, they received a lot of investment in the last few years with the introduction of v2 and lots of other changes. Check out https://www.rabbitmq.com/blog/2024/01/11/3.13-release for example.

What is deprecated is mirroring of classic queues. Which is something you don't need for RPC.
